os/arch/arm/src/imxrt/mpu-reg.h

```diff
@@ -0,0 +1,58 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+#ifndef __ARCH_ARM_SRC_IMXRT_IMXRT_MPUREG_H
+#define __ARCH_ARM_SRC_IMXRT_IMXRT_MPUREG_H
+
+/****************************************************************************
+ * Included FilesG
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <tinyara/irq.h>
+
+#ifndef __ASSEMBLY__
+#include <stdint.h>
+#include <arch/arch.h>
+#endif
+
+#if defined(CONFIG_ARMV7M_MPU) && defined(CONFIG_BUILD_PROTECTED)
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+/*
+ * Region definitions for our platform
+ */
+#define MPU_REG_KERN_CODE	MPU_REG0
+#define MPU_REG_KERN_DATA	MPU_REG1
+#define MPU_REG_USER_CODE	MPU_REG2
+#define MPU_REG_USER_DATA	MPU_REG3
```

Comment:
Should be they fixed?

Reply:
It is better to reserve region for robustness as we have enough regions, else it will create one more dependency in kernel for mapping and managing the regions.

Reply:
@sunghan-chang 
It is necessary to have fixed regions. 
When we add code for MPU in app separation scenario, the fixed regions becomes important. 
We will have a single region fixed for the app memory protection. Whenever there is a context switch, we need to update this MPU region with the new app's memory range.

Reply:
Then, could you leave comments to say necessary of this and why?

And one more thing.
Is this file, os/arch/arm/src/imxrt/mpu-reg.h chip-specific?
Can't we make it chip-common file?

Reply:
From your comment, I think they should be fixed regardless of chip or board. And it looks it does not have any chip-specific contents. Then why not common file?

Reply:
The 4 basic regions like kernel and user code and data are common to all chips. 
But there might be other regions which are specific to the chip. 
For example imxrt has dtcm, itcm and ocram which we might decide to use for some purpose. Then we need to configure separate regions for each of them. 
Other chips might not have these regions or might have some other memory regions to be configured. So, I think this is chip specific.

Reply:
yes it is chip specific, as I remember in broadcom case also there were reserved memory regions set. So the available regions might depend on chip

Reply:
That means those 4 regions can be another numbers. Anyway we need 4 fixed regions for them but number can be changed. Am I right?

Reply:
yes, depends on reserved regions for chip

Reply:
updated